src/serde_xml_plist.rs

```diff
@@ -1,467 +1,657 @@
+                Value::Boolean(true) => parent.serialize_field("true", &()),
+                Value::Boolean(false) => parent.serialize_field("false", &()),
+                Value::Data(_) => parent.serialize_field("data", &ValueInnerHelper(self.0)),
+                Value::Date(_) => parent.serialize_field("date", &ValueInnerHelper(self.0)),
+                Value::Real(_) => parent.serialize_field("real", &ValueInnerHelper(self.0)),
+                Value::Integer(_) => parent.serialize_field("integer", &ValueInnerHelper(self.0)),
+                Value::String(_) => parent.serialize_field("string", &ValueInnerHelper(self.0)),
+                v => Err(S::Error::custom(format_args!("unsupported plist::Value: {v:?}"))),
+            }
+        }
+    }
 
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
-            "key" => Ok(Self),
-            other => Err(other.to_string()),
+    impl Serialize for ValueHelper<'_> {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+        {
+            // XML uses field names (ignoring struct names) to generate tags, so a tagged value is
+            // serialized as a struct with a field of the tag name, the value being the inner
+            // e.g. `Value::Integer(10)` -> `struct { int: 10 }` -> `<int>10</int>`
+            let mut dummy = serializer.serialize_struct("", 1)?;
+            self.serialize_within::<S>(&mut dummy)?;
+            dummy.end()
         }
     }
-}
 
-impl FromStr for ValueKeyword {
-    type Err = String;
-
-    fn from_str(s: &str) -> Result<Self, Self::Err> {
-        match s {
-            "dict" => Ok(Self::Dict),
-            "array" => Ok(Self::Array),
-            "integer" => Ok(Self::Integer),
-            "real" => Ok(Self::Real),
-            "string" => Ok(Self::String),
-            "data" => Ok(Self::Data),
-            "date" => Ok(Self::Date),
-            "true" => Ok(Self::True),
-            "false" => Ok(Self::False),
-            other => Err(other.to_string()),
+    /// Serialises a value, without its surrounding tag
+    ///
+    /// e.g. `Value::Integer(10)` -> `10`
+    struct ValueInnerHelper<'lib>(&'lib Value);
+
+    impl Serialize for ValueInnerHelper<'_> {
+        fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+        {
+            match &self.0 {
+                Value::Array(a) => ArrayInnerHelper(a).serialize(serializer),
+                Value::Dictionary(d) => DictionaryInnerHelper(d).serialize(serializer),
+                Value::Boolean(_) => {
+                    unreachable!(
+                        "ValueInnerHelper should never serialize a boolean as it has no inner"
+                    )
+                }
+                Value::Data(d) => serializer.serialize_str(&base64_standard.encode(d)),
```

Comment:
when we get plist v1.6 on crates.io we'll want to replace this with `Data::to_xml_format`.